# Using Phylogenomic Data to Resolve the Hominoid Trichotomy

Abstract: The “hominoid trichotomy” is a classic issue in primate molecular systematics in which
the evolutionary relationships between humans, chimpanzees, and gorillas remain unresolved,
resulting in a polytomy at their ancestral node. Historically, immunological
studies succeeded in identifying chimpanzees and gorillas as our closest living relatives 
(Sarich and Wilson 1967). However, these early molecular studies using protein similarities
failed to recognize a Chimp-Human, Gorilla-Human, or Gorilla-Chimp clade. With the advent of moonshot 
DNA sequencing projects, it was revealed that humans share greater overall
sequence similarity with chimpanzees over gorillas. However, this similarity varies across the
genome with some chromosomes diverging less with chimpanzees, while remaining near the expected 
values for gorillas (Patterson et al., 2006). This project will employ publicly available 
orthologous gene datasets from the National Center for Biotechnology Information (NCBI) and will 
utilize an updated suite of bioinformatics tools to reexamine the hominoid trichotomy. This project 
will build different trees based of orthologous genomic segments for great apes (homininae) 
with orangutans (Pongo) as an outgroup.

![Figure 1. Historical development of hominid (great ape) systematics, figure adapted from Stoneking 2017](images/Figure1.Trees.png)

#### Outline

* [1. Downloading NCBI Gene Data](#section-1-Downloading-NCBI-Gene-Data)
* [2. NCBI Datasets Command Line Tools](#section-2-NCBI-Datasets-Command-Line-Tools)
* [3. MAFFT Alignment](#section-3-MAFFT-Alignment)
* [4. RAXML-NG Analysis](#section-4-RAXML-NG-Analysis)

## Section 1: Downloading NCBI Gene Data

## Section 2: NCBI Datasets Command Line Tools

## Section 3: MAFFT Alignment

## Section 4: RAXML NG Analysis
